lua/nvim-tree.lua

```diff
@@ -336,6 +336,8 @@ local function setup_autocommands(opts)
   if opts.hijack_directories.enable then
     create_nvim_tree_autocmd({ "BufEnter", "BufNewFile" }, { callback = M.open_on_directory })
   end
+
+  create_nvim_tree_autocmd("BufEnter", { pattern = "NvimTree_*", callback = reloaders.reload_explorer })
```

Comment:
I wouldn't do that by default, some people have large trees and reloading is a bit expensive. This should be hidden behind a config option at least :) something like `reload_on_bufenter`, which should be falsy by default.

Reply:
Agreed.

That said, this is an option I will probably turn on, as I do find myself refreshing after external changes / make etc.

Reply:
Added